test/e2e/tests/transaction/change-assets.spec.js

```diff
@@ -342,7 +342,7 @@ describe('Change assets', function () {
 
         // Make sure gas is updated by resetting amount and hex data
         // Note: this is needed until the race condition is fixed on the wallet level (issue #25243)
-        await driver.fill('[data-testid="currency-input"]', '2');
+        await driver.fill('[data-testid="currency-input"]', '2.000042');
```

Comment:
@hjetpoluru has this issue been resolved? This is unrelated to the changes in this PR, but this change was necessary in order to get this passing.

Could you suggest if this should be included? Or if there is another fix?

Reply:
Will this failing unit test block this PR from getting merged? Since the application bug is unrelated to these change, can we just `.skip` this test?

Reply:
If you're lucky this e2e test should not block you from merging. We should not skip any tests and should prioritize addressing the bug.

I noticed that there is a failing unit test, and you may need to check and fix it

Reply:
Still cleaning up the code, but it's green and has been passing recently. Should I reach out again once this is ready for formal review?